tests/test_expval.py

```diff
@@ -38,15 +38,8 @@ def test_identity_expectation(self, theta, phi, qubit_device, tol):
         O1 = qml.Identity(wires=[0])
         O2 = qml.Identity(wires=[1])
         ops = [qml.RX(theta, wires=[0]), qml.RX(phi, wires=[1]), qml.CNOT(wires=[0, 1])]
-        if ld._new_API:
-            tape = qml.tape.QuantumScript(ops, [qml.expval(O1), qml.expval(O2)])
-            res = dev.execute(tape)
-        else:
-            dev.apply(
-                ops,
-                rotations=[*O1.diagonalizing_gates(), *O2.diagonalizing_gates()],
-            )
-            res = np.array([dev.expval(O1), dev.expval(O2)])
+        tape = qml.tape.QuantumScript(ops, [qml.expval(O1), qml.expval(O2)])
+        res = dev.execute(tape)
```

Comment:
Is there any remaining case where we would need ``ld._new_API``? If not, could you please remove this attribute definition?

Reply:
I wouldn't know, but this would be outside the scope of this PR. I simply made these changes as they were trivial.
I believe a more comprehensive cleanup should be done in a separate PR after the Facade PR is merged.
@maliasadi

Reply:
There are many more use cases for `ld._new_API` which will be removed eventually after migrating both LGPU and LK to the new device API

Reply:
Thank you for clarifying.